### PR TITLE
refactor: Remove unused function add and unused variable keysHint

### PR DESCRIPTION
**Pull request by lesam**
Very small refactor so no tracking issue

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass
- [x] http/swagger.yml updated (if modified Go structs or API)
- [ ] Feature flagged (if modified API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: tsdb/engine/tsm1/ring.go [hunk 1]
```diff
@@ -103,28 +95,16 @@ func (r *ring) write(key []byte, values Values) (bool, error) {
 	return r.getPartition(key).write(key, values)
 }
 
-// add adds an entry to the ring.
-func (r *ring) add(key []byte, entry *entry) {
-	r.getPartition(key).add(key, entry)
-	atomic.AddInt64(&r.keysHint, 1)
```
Comment (lesam) [inline tsdb/engine/tsm1/ring.go:109]: Note that since add is removed (it was unused in production code), keysHint is always 0

**Comment by e-dard** [comment]
This seems fine to me correctness wise but I'm surprised that the `ring.add` call is never used. I would be even more surprised if it _has_ never been used, rather than being removed through some refactor.

The original reason I added these sizing hints was to improve `tsm1.Cache` snapshotting and write performance by reducing allocations. Typically in a running system the same type of data gets continuously written and so if you pre-allocate bits of the cache in the same way after a snapshot you can minimise the number of allocations.

If `add` was just removed as a refactor somewhere I wonder about putting it back so that `keysHint` can become useful again. Having said that, a lot changes in four years and the cache as it runs today may not benefit from these pre-allocation hints anyway 🤷

**Review by e-dard — COMMENTED** [review]
This seems fine to me correctness wise but I'm surprised that the ring.add call is never used. I would be even more surprised if it has never been used, rather than being removed through some refactor.

The original reason I added these sizing hints was to improve tsm1.Cache snapshotting and write performance by reducing allocations. Typically in a running system the same type of data gets continuously written and so if you pre-allocate bits of the cache in the same way after a snapshot you can minimise the number of allocations.

If add was just removed as a refactor somewhere I wonder about putting it back so that keysHint can become useful again. Having said that, a lot changes in four years and the cache as it runs today may not benefit from these pre-allocation hints anyway 🤷

**Review by dgnorton — APPROVED** [review]

**Comment by luismedicina23** [comment]
@Main
- Remove

**Pull request merged**